plugins/out_stackdriver/stackdriver.c

```diff
@@ -494,35 +546,22 @@ static int stackdriver_format(const void *data, size_t bytes,
     msgpack_pack_str(&mp_pck, 6);
     msgpack_pack_str_body(&mp_pck, "labels", 6);
 
-    if (strcmp(ctx->resource, "global") == 0) {
-      /* global resource has field project_id */
-      msgpack_pack_map(&mp_pck, 1);
-      msgpack_pack_str(&mp_pck, 10);
-      msgpack_pack_str_body(&mp_pck, "project_id", 10);
-      msgpack_pack_str(&mp_pck, flb_sds_len(ctx->project_id));
-      msgpack_pack_str_body(&mp_pck,
-                            ctx->project_id, flb_sds_len(ctx->project_id));
-    }
-    else if (strcmp(ctx->resource, "gce_instance") == 0) {
-      /* gce_instance resource has fields project_id, zone, instance_id */
-      msgpack_pack_map(&mp_pck, 3);
-
-      msgpack_pack_str(&mp_pck, 10);
-      msgpack_pack_str_body(&mp_pck, "project_id", 10);
-      msgpack_pack_str(&mp_pck, flb_sds_len(ctx->project_id));
-      msgpack_pack_str_body(&mp_pck,
-                            ctx->project_id, flb_sds_len(ctx->project_id));
-
-      msgpack_pack_str(&mp_pck, 4);
-      msgpack_pack_str_body(&mp_pck, "zone", 4);
-      msgpack_pack_str(&mp_pck, flb_sds_len(ctx->zone));
-      msgpack_pack_str_body(&mp_pck, ctx->zone, flb_sds_len(ctx->zone));
-
-      msgpack_pack_str(&mp_pck, 11);
-      msgpack_pack_str_body(&mp_pck, "instance_id", 11);
-      msgpack_pack_str(&mp_pck, flb_sds_len(ctx->instance_id));
-      msgpack_pack_str_body(&mp_pck,
-                            ctx->instance_id, flb_sds_len(ctx->instance_id));
+    /*
+     * Iterate through the defined labels for this resource.
+     *
+     * global       - project_id
+     * gce_instance - project_id, instance_id, zone
+     * generic_node - project_id, location, namespace, node_id
+     * generic_task - project_id, location, namespace, job, task_id
+     */
+    msgpack_pack_map(&mp_pck, mk_list_size(&ctx->resource_labels));
+    mk_list_foreach(label, &ctx->resource_labels) {
+        kv = mk_list_entry(label, struct flb_kv, _head);
+        msgpack_pack_str(&mp_pck, flb_sds_len(kv->key));
+        msgpack_pack_str_body(&mp_pck, kv->key, flb_sds_len(kv->key));
+        msgpack_pack_str(&mp_pck, flb_sds_len(kv->val));
+        msgpack_pack_str_body(&mp_pck,
+                              kv->val, flb_sds_len(kv->val));
```

Comment:
This easily fits on a single line. Any reason you split it over two?

Reply:
Nope, must have been an auto-formatting quirk.